index.js

```diff
@@ -46,12 +62,36 @@ module.exports = fp(async function (app, { all, policy, ttl, cacheSize, skip, st
       cacheSize
     })
   }
+
+  function initCacheReport () {
+    cacheReport = {}
+    for (const field of Object.keys(policy.Query)) {
+      const name = 'Query.' + field
+      cacheReport[name] = {}
+      cacheReport[name].hits = 0
+      cacheReport[name].misses = 0
+    }
+  }
+
+  function clearCacheReport () {
+    if (!cacheReport) return
+
+    for (const item of Object.keys(cacheReport)) {
+      cacheReport[item].hits = 0
+      cacheReport[item].misses = 0
+    }
+  }
+
+  function logReport () {
+    app.log.info(`Cache report - ${JSON.stringify(cacheReport)}`)
+    console.table(cacheReport)
```

Comment:
should we allow the user to specify a custom function to build the report output or is this ok for now?

Also, I could only think of `console.table` for building the table report. Open to any suggestions to replace it.

Reply:
You should not use console at all. Just dump the object to the loggger